llvm/include/llvm/RemoteCachingService/Client.h

```diff
@@ -61,6 +62,35 @@ class KeyValueDBClient {
 
   using ValueTy = StringMap<std::string>;
 
+  struct GetValueResType {
+    std::shared_ptr<AsyncCallerContext> CallCtx;
+    // If this is \p None it means the key was not found.
+    Optional<ValueTy> Value;
+  };
+  struct PutValueResType {
+    std::shared_ptr<AsyncCallerContext> CallCtx;
+  };
+  Expected<GetValueResType> getValueSync(std::string Key) {
```

Comment:
It'd be better to separate the return values of the sync APIs from the async ones. `std::shared_ptr<AsyncCallerContext> CallCtx` has no value being in the return structure of a sync API, you can simplify the sync APIs and make them more convenient to use by returning the resulting values directly; for example:
```
Expected<Optional<ValueTy> Value> getValueSync(std::string Key);
Error putValueSync(std::string Key, const ValueTy &Value)
```